src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -467,4 +484,82 @@ public List<Project> getProjects(Vulnerability vulnerability) {
+        }
+        if (alias.getOsvId() != null) {
+            if (filter.length() > 0) filter += " && ";
+            filter += "(osvId == :osvId || osvId == null)";
+            params.put("osvId", alias.getOsvId());
+        }
+        if (alias.getGsdId() != null) {
+            if (filter.length() > 0) filter += " && ";
+            filter += "(gsdId == :gsdId || gsdId == null)";
+            params.put("gsdId", alias.getGsdId());
+        }
+        if (alias.getVulnDbId() != null) {
+            if (filter.length() > 0) filter += " && ";
+            filter += "(vulnDbId == :vulnDbId || vulnDbId == null)";
+            params.put("vulnDbId", alias.getVulnDbId());
+        }
+        if (alias.getInternalId() != null) {
+            if (filter.length() > 0) filter += " && ";
+            filter += "(internalId == :internalId || internalId == null)";
+            params.put("internalId", alias.getInternalId());
+        }
+        final Query<VulnerabilityAlias> query = pm.newQuery(VulnerabilityAlias.class);
+        query.setFilter(filter);
+        final VulnerabilityAlias existingAlias = singleResult(query.executeWithMap(params));
+        if (existingAlias != null) {
+            // Check if each of the fields are null. We don't want to overwrite an alias where only partial information is known, with nulls.
+            if (alias.getCveId() != null) existingAlias.setCveId(alias.getCveId());
+            if (alias.getSonatypeId() != null) existingAlias.setSonatypeId(alias.getSonatypeId());
+            if (alias.getGhsaId() != null) existingAlias.setGhsaId(alias.getGhsaId());
+            if (alias.getSnykId() != null) existingAlias.setSnykId(alias.getSnykId());
+            if (alias.getOsvId() != null) existingAlias.setOsvId(alias.getOsvId());
+            if (alias.getGsdId() != null) existingAlias.setGsdId(alias.getGsdId());
+            if (alias.getVulnDbId() != null) existingAlias.setVulnDbId(alias.getVulnDbId());
+            if (alias.getInternalId() != null) existingAlias.setInternalId(alias.getInternalId());
+            return persist(existingAlias);
+        } else {
+            return persist(alias);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<VulnerabilityAlias> getVulnerabilityAliases(Vulnerability vulnerability) {
+        final Query<VulnerabilityAlias> query;
+        if (Vulnerability.Source.NVD.name().equals(vulnerability.getSource())) {
+            query = pm.newQuery(VulnerabilityAlias.class, "cveId == :cveId");
+        } else if (Vulnerability.Source.OSSINDEX.name().equals(vulnerability.getSource())) {
+            query = pm.newQuery(VulnerabilityAlias.class, "sonatypeId == :sonatypeId");
+        } else if (Vulnerability.Source.GITHUB.name().equals(vulnerability.getSource())) {
+            query = pm.newQuery(VulnerabilityAlias.class, "ghsaId == :ghsaId");
+        } else if (Vulnerability.Source.VULNDB.name().equals(vulnerability.getSource())) {
```

Comment:
Need addition of osv id as well for google datasource:
`else if (Vulnerability.Source.OSV.name().equals(vulnerability.getSource())) {
            query = pm.newQuery(VulnerabilityAlias.class, "osvId == :osvId");
        } `

Reply:
Good catch. Resolved in https://github.com/DependencyTrack/dependency-track/pull/1912/commits/9d200158d2c6686d56ba74f6d15b34f88f550060